clang/include/clang/CConv/ConstraintVariables.h

```diff
@@ -282,6 +287,10 @@ class PointerVariableConstraint : public ConstraintVariable {
   // should be modified during rewriting.
   bool OriginallyChecked;
 
+  bool IsTypedef = false;
+  TypedefNameDecl* TDT;
+  std::string typedefString;
```

Comment:
Add comment to explain these. Do we need `IsTypedef`? That is, is it always the case that `TDT == nullptr` if and only if `IsTypedef == false` ? If so, init `TDT` to `nullptr` and adjust accordingly?